pkg/tsdb/influxdb/response_parser.go

```diff
@@ -56,32 +56,55 @@ func parseJSON(buf io.ReadCloser) (Response, error) {
 func transformRows(rows []Row, query Query) data.Frames {
 	frames := data.Frames{}
 	for _, row := range rows {
-		for columnIndex, column := range row.Columns {
+		var hasTimeCol = false
+
+		for _, column := range row.Columns {
 			if column == "time" {
-				continue
+				hasTimeCol = true
 			}
+		}
 
-			var timeArray []time.Time
-			var valueArray []*float64
+		if !hasTimeCol {
+			var values []string
 
 			for _, valuePair := range row.Values {
-				timestamp, timestampErr := parseTimestamp(valuePair[0])
-				// we only add this row if the timestamp is valid
-				if timestampErr == nil {
-					value := parseValue(valuePair[columnIndex])
-					timeArray = append(timeArray, timestamp)
-					valueArray = append(valueArray, value)
+				if strings.Contains(query.RawQuery, "SHOW TAG VALUES") {
+					values = append(values, valuePair[1].(string))
```

Comment:
@joey-grafana - can we handle this as case-insensitive? 

nit: Is there any chance the actual data can not be in index 1. 🤔